x-pack/plugins/apm/public/components/shared/service_link.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink } from '@elastic/eui';
+import React from 'react';
+import { euiStyled } from '../../../../../../src/plugins/kibana_react/common';
+import { truncate } from '../../utils/style';
+import { useApmRouter } from '../../hooks/use_apm_router';
+import { AgentIcon } from './agent_icon';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+
+const StyledLink = euiStyled(EuiLink)`${truncate('100%')};`;
```

Comment:
Maybe this could have its own Property? You're using it on BackendLink too

Reply:
The two components are very similar. I might put them together into a `NodeLink` or something once #106995 is merged.